pkg/kratos/handlers.go

```diff
@@ -199,17 +203,69 @@ func (a *API) handleUpdateFlow(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	shouldRegenerateBackupCodes, err := a.shouldRegenerateBackupCodes(r.Context(), cookies)
+	if err != nil {
+		err = fmt.Errorf("error when checking backup codes: %v", err)
+
+		a.logger.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
 	setCookies(w, cookies)
 
 	if shouldEnforceMfa {
```

Comment:
not part of this pr, but there are not tests covering this if clause

mind adding an issue in gh?

Reply:
Sure
https://github.com/canonical/identity-platform-login-ui/issues/271